modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/StrictWFS_2_0_Strategy.java

```diff
@@ -662,24 +665,48 @@ protected OperationType getOperationMetadata(final WFSOperationType operation) {
         return ftypeCrss;
     }
 
+    /**
+     * Returns parameters defined in OperationsMetadata
+     *
+     * @param parameterName
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    private Set<String> findParameters(final String parameterName) {
+        final OperationsMetadataType operationsMetadata = capabilities.getOperationsMetadata();
+
+        List<DomainType> parameters = operationsMetadata.getParameter();
+        for (DomainType parameter : parameters) {
+            if (parameterName.equals(parameter.getName())) {
+                Set<String> outputFormats = new HashSet<>();
+                for (ValueType value : (List<ValueType>) parameter.getAllowedValues().getValue()) {
+                    outputFormats.add(value.getValue());
+                }
+                return outputFormats;
+            }
+        }
+        return Collections.emptySet();
+    }
+
     @SuppressWarnings("unchecked")
     protected Set<String> findParameters(
             final OperationType operationMetadata, final String parameterName) {
-        Set<String> outputFormats = new HashSet<>();
 
         List<DomainType> parameters = operationMetadata.getParameter();
         for (DomainType param : parameters) {
 
             String paramName = param.getName();
 
             if (parameterName.equals(paramName)) {
+                Set<String> outputFormats = new HashSet<>();
```

Comment:
Same as above (I see it was already like that, but it's a rename-refactor away from being fixed).

Reply:
I've renamed to foundValues.